
<a href='https://www.darshan.ac.in/'> <img src='https://www.darshan.ac.in/Content/media/DU_Logo.svg' width="250" height="300"/></a>
<pre>
<center><b><h1>Python Programming - 2301CS404</b></center>
<center><b><h1>Lab - 4</b></center>   
<center><b><h1>Raj Panchasara | 23010101185 | A8</b></center>
<pre>

# String

### 01) WAP to check whether the given string is palindrome or not.

In [11]:
a = input("Enter String : ")

b = a[::-1]

if(a==b):
    print("palindrome")
else :
    print("Not palindrome")

Enter String : 121
palindrome


### 02) WAP to reverse the words in the given string.

In [21]:
a = input("Enter String : ")
l1 = a.split(" ")
l2 = l1[::-1]
b = ' '.join(l2)

print('Reversed String : \b',b)

Enter String : 123 456 789
Reversed String : 789 456 123


### 03) WAP to remove ith character from given string.

In [26]:
a = input("Enter String : ")
i = int(input("Enter index : "))

l1 = a[:i:] + a[i+1::]
print(l1)

Enter String : abcde
Enter index : 2
abde


### 04) WAP to find length of string without using len function.

In [31]:
a = input("Enter String : ")
b = 0
for i in a:
    b += 1
print(b)

Enter String : 123456789
9


### 05) WAP to print even length word in string.

In [34]:
a = input("Enter String : ")
l1 = a.split(' ')

for i in l1:
    b = 0
    for j in i:
        b += 1
    if(b%2==0):
        print(i+' ',end="")

Enter String : 1234 123 12345 123456
1234 123456 

### 06) WAP to count numbers of vowels in given string.

In [37]:
a = input("Enter String : ")
b = 0 

#  'a' in  i
#   i  in 'a'
#   i == 'a'
for i in a:
    if(i in 'a' or i=='e' or i=='i' or i=='o' or i=='u' or i in 'A' or i=='E' or i=='I' or i=='O' or i=='U'):
        b += 1
print(b)

Enter String : raj PANCHASARA
5


### 07) WAP to capitalize the first and last character of each word in a string.

In [17]:
def capitalize(a):
    return ' '.join( word[0].upper() + word[1:-1].lower() + word[-1].upper() if len(word)>1 else word.upper()
                     for word in a.split())

a = "hello world from pytho n"
b = capitalize(a)
print(b)

HellO WorlD FroM PythO N


### 08) WAP to convert given array to string.

In [1]:
l1 = []
for i in range(1,6) :
     l1.append(int(input('Enter element : ')))

a = ''
for i in l1 :
    a += str(i)

# OR a = ''.join(str(l1)) ======> a = [1,2,3,4,5]
print('Converted string = ', a) 

Enter element :  1
Enter element :  2
Enter element :  3
Enter element :  4
Enter element :  5


Converted string =  12345


### 09) Check if the password and confirm password is same or not. 
### In case of only case's mistake, show the error message.

In [3]:
p1 = input('Enter Password :')
p2 = input('Enter confirm password :')

if(p1 == p2) :
    print('Passwords match')
elif(p1.lower() == p2.lower()) :
    print('Case mistake')
else :
    print('Passwords do not match')

Enter Password : qwer@123A
Enter confirm password : qwer@123A


Passwords match


### 10) : Display credit card number. 
### card no. : 1234 5678 9012 3456
### display as  : **** **** **** 3456

In [4]:
a = input('Enter the card number : ')
l1 = a.split(' ')
a = '**** ' * 3 + l1[3]
print('Masked card number : ', a)

Enter the card number :  1234 5678 9012 3456


Masked card number :  **** **** **** 3456


### 11) : Checking if the two strings are Anagram or not.
### s1 = decimal and s2 = medical are Anagram

In [5]:
str1 = input('Enter string 1 : ')
str2 = input('Enter string 2 : ')

if sorted(str1) == sorted(str2) :
    print('Anagram')
else :
    print('Not anagram')

Enter string 1 :  raj
Enter string 2 :  jar


Anagram


### 12) : Rearrange the given string. First lowercase then uppercase alphabets.
### input : EHlsarwiwhtwMV
### output : lsarwiwhtwEHMV

In [6]:
str1 = input('Enter the string : ')
lower = ''
upper = ''
for i in str1 :
    if i.islower() :
        lower += i
    else :
        upper += i

str1 = lower + upper
print(str1)

Enter the string :  EHlsarwiwhtwMV


lsarwiwhtwEHMV


In [ ]:
#-------------------------------------------------------------------------------------------------------------------------------

In [1]:
name = "raj"
age = 19

print("Name: " + name + ", Age: " + str(age))        # Using Concatenation (+) : ( String only )

print("Name:" , name , ", Age:" ,age)                # Using Comma-separated Values : ( Automatically handles type and adds a space between arguments.)

print("Name: " + name + ", Age: " ,age)              # Using Concatenation and Comma-separated

print("Name: %s, Age: %d" % (name, age))             # Using % Formatting: (Requires specifying the type (%s for string, %d for int, %f for float).)

print("Name: {}, Age: {}".format(name, age))         # Using .format() Method: (Automatically converts types and uses placeholders ({}).)
print("Age: {1}, Name: {0}".format(name, age))       # With Positional Arguments:
print("Name: {n}, Age: {a}".format(n=name, a=age))   # With Named Arguments:

print(f"Name: {name} , Age: {age}")                  # Using f-Strings (Python 3.6+):( The most modern and readable method. Allows inline expressions)
print(f"Name: {name.upper()}, Age: {age + 5}")       # can use method too

print("Name: " + repr(name) + ", Age: " + repr(age))

Name: raj, Age: 19
Name: raj , Age: 19
Name: raj, Age:  19
Name: raj, Age: 19
Name: raj, Age: 19
Age: 19, Name: raj
Name: raj, Age: 19
Name: raj , Age: 19
Name: RAJ, Age: 24
Name: 'raj', Age: 19


In [ ]:
"""
| Method                  | Advantages                       | Disadvantages                |
|-------------------------|----------------------------------|------------------------------|
| Concatenation           | Simple for strings               | Manual type conversion       |
| Comma-separated         | Handles types automatically      | Adds unwanted spaces         |
| `%` Formatting          | Compact, widely used             | Verbose for many variables   |
| `.format()` Method      | Flexible and readable            | A bit verbose                |
| f-Strings               | Concise, readable, and modern    | Requires Python 3.6+         |
| String Templates        | Great for external templates     | Less commonly used           |

Choose the method based on readability, maintainability, and project requirements!
"""